web/src/utils/factory-management/syncState.spec.ts

```diff
@@ -238,6 +238,82 @@ describe('syncState', () => {
         calculateSyncState(mockFactory)
         expect(mockFactory.inSync).toBe(true)
       })
+
+      it('should remain in sync for fuel-only factories (no products, only power producers)', () => {
+        // Create a fuel-only factory (coal power plant scenario)
+        const fuelOnlyFactory = newFactory('Coal Power Plant')
+
+        // Add only power producer, no products
+        addPowerProducerToFactory(fuelOnlyFactory, {
+          building: 'generatorfuel',
+          buildingAmount: 1,
+          recipe: 'GeneratorFuel_Coal',
+          updated: 'power',
+        })
+
+        // Set sync state
+        setSyncState(fuelOnlyFactory)
+        expect(fuelOnlyFactory.inSync).toBe(true)
+        expect(fuelOnlyFactory.products.length).toBe(0)
+        expect(fuelOnlyFactory.powerProducers.length).toBe(1)
+
+        // Calculate sync state - should remain in sync since power configuration hasn't changed
+        calculateSyncState(fuelOnlyFactory)
+        expect(fuelOnlyFactory.inSync).toBe(true)
+      })
+
+      it('should go out of sync for empty factories (no products, no power producers)', () => {
+        // Create an empty factory
+        const emptyFactory = newFactory('Empty Factory')
+
+        // Set sync state first (simulating it was previously in sync)
+        emptyFactory.inSync = true
+        emptyFactory.syncState = {}
+        emptyFactory.syncStatePower = {}
+
+        expect(emptyFactory.inSync).toBe(true)
+        expect(emptyFactory.products.length).toBe(0)
+        expect(emptyFactory.powerProducers.length).toBe(0)
+
+        // Calculate sync state - should go out of sync since factory is truly empty
+        calculateSyncState(emptyFactory)
+        expect(emptyFactory.inSync).toBe(false)
```

Comment:
Ideally, we shouldn't be able to set a factory as "in sync" at all if there's nothing to sync, but that's a slightly different, arguably non-issue.